api-metastore/src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -398,34 +398,42 @@ public void delete(final String eventTypeName) throws EventTypeDeletionException
 
     private Multimap<TopicRepository, String> deleteEventTypeWithSubscriptions(final String eventType) {
         try {
-            return transactionTemplate.execute(action -> {
-                SubscriptionTokenLister.ListResult listResult = subscriptionTokenLister.listSubscriptions(
-                        ImmutableSet.of(eventType), Optional.empty(), Optional.empty(), null, 100);
-                while (null != listResult) {
-                    listResult.getItems().forEach(s -> {
-                        try {
-                            subscriptionRepository.deleteSubscription(s.getId());
-                        } catch (final NoSuchSubscriptionException e) {
-                            // should not happen as we are inside transaction
-                            throw new InconsistentStateException("Subscription to be deleted is not found", e);
+            return eventTypeRepository.lockingTable(EventTypeRepository.TableLockMode.ROW_EXCLUSIVE,
+                    transactionTemplate,
+                    action -> {
```

Comment:
Then `action` at this point should also be named `status`, or not?

Reply:
forgot abt this correct, the whole Lambda is the action but the parameter is status.